packages/liveblocks-core/src/room.ts

```diff
@@ -1341,7 +1342,7 @@ export function createRoom<
     const stackSizeBefore = context.undoStack.length;
     for (const key in context.initialStorage) {
       if (context.root.get(key) === undefined) {
-        context.root.set(key, context.initialStorage[key]);
+        context.root.set(key, cloneLson(context.initialStorage[key]));
```

Comment:
@GuillaumeSalles What do you think of doing this when initializing `initialStorage`? This would avoid the error from #1000 from ever showing up, even if people set it up using a global instance accidentally. I think this is pretty nice, but not sure if maybe you see a caveat with this?

Reply:
We just chatted about this, and decided to change this slightly to the following behavior:

- We do a quick check to see if `context.initialStorage[key]` is a Live structure
  - If so, we check to see if it has a parent node set (i.e. it's part of a tree already)
     - If it has a parent, we `.clone()` it, and issue a warning that this can be indicative of a bug in the way the app is configured, and that the user should always use fresh/new instances
     - If not, great, just use it
  - If not, it's a JSON value, so no need to clone anything

Reply:
Will implement this next week.

Reply:
Now that I started implementing this, I'm having second thoughts about this solution, because doing the optional cloning leads to situations that are maybe even more confusing, potentially.

Here's such a footgun:

- First, empty/new room A is entered with `initialStorage={initialStorage}`
- Keys are missing, so the global `initialStorage` values are assigned, no cloning
- User now just uses the room, mutating things as they go, populating the document with whatever content. Since no clone wat taken, this will also mutate the global `initialStorage` instance (unintentionally)
- Now, the user joins empty/new room B with `initialStorage={initialStorage}`
- Keys are missing, but since global `initialStorage` values are detected to already have a parent, a clone is taken, but… the clone will now contain all the mutations made in room A!

This makes me conclude that the optional cloning tactic is maybe the worst solution even.

Therefore, I think that always cloning (i.e. what this PR implemented) is still be best pragmatic solution that leads to the fewest surprises — and reflects best how the prop is intended. We can document that whatever you pass as `initialStorage` will be considered a "template".

cc @GuillaumeSalles